src/main/scala/lambdaone/kratia/registry/Registry.scala

```diff
@@ -0,0 +1,41 @@
+package lambdaone.kratia.registry
+
+import cats.Monad
+import lambdaone.toolbox.{EventStore, Store}
+import cats.implicits._
+
+trait Registry[F[_], A] {
+
+  def isMember(community: Community[A], member: Member[A]): F[Boolean]
+
+  def load(community: Community[A], member: Member[A]): F[Option[A]]
+
+  def loadAll(community: Community[A]): F[List[A]]
+
+  def register(community: Community[A], member: Member[A], data: A): F[()]
+}
+
+object Registry {
+
+  implicit def registryWithEventStore[F[_], A](implicit store: EventStore[F, RegistryEvent[A]], F: Monad[F]): Registry[F, A] =
+    new Registry[F, A] {
+
+      override def isMember(community: Community[A], member: Member[A]): F[Boolean] =
+        store.find {
+          case RegistryEvent.RegisterMember(community0, member0, _) => community == community0 && member == member0
+        }.map(_.isDefined)
+
+      override def load(community: Community[A], member: Member[A]): F[Option[A]] =
+        store.findMap {
+          case RegistryEvent.RegisterMember(community0, member0, data) if community == community0 && member == member0 => data
```

Comment:
Event Store easily leads to CQRS. In a slightly more realistic scenario, we would need to go through literally all events in the registry (UnregisterMember, setData[A](a:A) etc.

Reply:
Indeed, with CQRS we have a "write optimized store" and a "read optimized store", the write is basically the event store, and the read is a store that keeps the latest state in maybe a denormalized form for performase, do you think we should reflect this as well? It might be a good idea